packages/compass-e2e-tests/smoke-test.ts

```diff
@@ -149,186 +159,91 @@ async function run() {
 
   assertCommonBuildInfo(buildInfo);
 
-  // filter the extensions given the platform (isWindows, isOSX, isUbuntu, isRHEL) and extension
-  const { isWindows, isOSX, isRHEL, isUbuntu, extension } = context;
-
-  const packages = getFilteredPackages(compassDir, buildInfo, {
-    isWindows,
-    isOSX,
-    isRHEL,
-    isUbuntu,
-    extension,
-  });
-
-  if (!context.skipDownload) {
-    await Promise.all(
-      packages.map(async ({ filename, filepath }) => {
-        await fs.mkdir(path.dirname(filepath), { recursive: true });
-        const url = `https://${context.bucketName}.s3.amazonaws.com/${context.bucketKeyPrefix}/${filename}`;
-        console.log(url);
-        return downloadFile(url, filepath);
-      })
-    );
-  }
-
-  verifyPackagesExist(packages);
-
-  // TODO(COMPASS-8533): extract or install each package and then test the Compass binary
-  for (const pkg of packages) {
-    let appInfo: InstalledAppInfo | undefined = undefined;
-
-    console.log('installing', pkg.filepath);
-
-    if (pkg.filename.endsWith('.dmg')) {
-      appInfo = await installMacDMG(buildInfo.productName, pkg);
-    }
-
-    // TODO: all the other installers go here
-
-    if (appInfo) {
-      console.log('testing', appInfo.appPath);
-      await testInstalledApp(appInfo);
-    } else {
-      console.log(`no app got installed for ${pkg.filename}`);
-    }
-  }
-}
-
-function platformFromContext(
-  context: SmokeTestsContext
-): typeof process.platform {
-  if (context.isWindows) {
-    return 'win32';
-  }
-
-  if (context.isOSX) {
-    return 'darwin';
-  }
-
-  if (context.isRHEL || context.isUbuntu) {
-    return 'linux';
-  }
-
-  return process.platform;
-}
-
-type PackageFilterConfig = Pick<
-  SmokeTestsContext,
-  'isWindows' | 'isOSX' | 'isRHEL' | 'isUbuntu' | 'extension'
->;
-
-// subsets of the hadron-build info result
-
-const commonKeys = ['productName'];
-type CommonBuildInfo = Record<typeof commonKeys[number], string>;
-
-function assertCommonBuildInfo(
-  buildInfo: unknown
-): asserts buildInfo is CommonBuildInfo {
-  assert(
-    typeof buildInfo === 'object' && buildInfo !== null,
-    'Expected buildInfo to be an object'
-  );
-  for (const key of commonKeys) {
-    assert(key in buildInfo, `Expected buildInfo to have '${key}'`);
+  let match:
+    | { filename: string; installer: Installer; updatable: boolean }
+    | undefined = undefined;
+
+  if (context.package === 'windows_setup') {
```

Comment:
This could also be a giant switch case or a hashmap to map these things or pluggable or.. I don't personally have a preference.